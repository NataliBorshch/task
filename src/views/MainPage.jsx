import './MainPage.scss';

const MainPage = () => {
  return (
    <div className="main_page">
      <div className="main_advice">
        <h2>10 ways to make your day as productive as possible</h2>
        <ol>
          <li>
            <p>Every Sunday night, make plans for the week</p>
          </li>
          <li>
            <p>Set aside time in advance for specific tasks</p>
          </li>
          <li>
            <p>Make sure the list of tasks is achievable</p>
          </li>
          <li>
            <p>Set up a 30-minute default meeting</p>
          </li>
          <li>
            <p>Say no multitasking</p>
          </li>
          <li>
            <p>Try to make the best use of «border» time</p>
          </li>
          <li>
            <p>Watch the time</p>
          </li>
          <li>
            <p>Think about dinner</p>
          </li>
          <li>
            <p>
              Don’t take up any of the time you have to spend with your family
            </p>
          </li>
          <li>
            <p>Start every day right</p>
          </li>
        </ol>
      </div>
      <div className="main_img">
        <img
          src="https://ik.imagekit.io/s2fpg15d4rx/time_NifncTLm-ND.jpg"
          alt="time"
        />
        <img
          src="https://ik.imagekit.io/s2fpg15d4rx/glasses1_4_fSdw54Wgo.jpg"
          alt="glasses"
        />
      </div>
    </div>
  );
};

export default MainPage;
