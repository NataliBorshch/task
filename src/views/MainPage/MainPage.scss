.main_page {
  display: flex;
  font-family: 'Satisfy';
}

h2 {
  font-size: 25px;
  font-family: 'Satisfy';
  text-transform: uppercase;
  word-wrap: break-word;
}

.main_advice {
  padding: 10px 20px;
}

.main_img {
  max-width: 100%;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  width: calc(100% / 2);
}
img {
  display: block;
  max-width: 500px;
}
