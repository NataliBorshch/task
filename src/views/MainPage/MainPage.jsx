import { connect } from 'react-redux';
import { Component } from 'react';
import { Switch, Route } from 'react-router';
// component
import TaskPage from '../../components/TaskPage';
import Panel from '../../components/Panel';
import Stats from '../../components/Stats';
import Profile from '../../components/Profile';
import TaskInfo from '../../components/TaskInfo/TaskInfo';
// redux
import usersOperations from '../../redux/users/operations-user';
import taskOperations from '../../redux/task/operations';
// styles
import './MainPage.scss';

class MainPage extends Component {
  componentDidMount() {
    this.props.getCurrentUser();
    this.props.getAllTask();
  }
  render() {
    return (
      <div>
        <div className="page_main">
          <div>
            <TaskInfo />
            <Panel />
          </div>
          <div>
            <Switch>
              <Route exact path="/home/tasks" component={TaskPage} />
              <Route exact path="/home/profile" component={Profile} />
              <Route exact path="/home/stats" component={Stats} />
            </Switch>
          </div>
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  getCurrentUser: () => dispatch(usersOperations.getCurrentUser()),
  getAllTask: () => dispatch(taskOperations.getTask()),
});

export default connect(null, mapDispatchToProps)(MainPage);
