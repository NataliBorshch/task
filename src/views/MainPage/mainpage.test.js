import { render, screen } from '@testing-library/react';
import MainPage from './MainPage';

describe('MainPage component ', () => {
  it('MainPage render', () => {
    render(<MainPage />);
    expect(screen.getByRole('list')).toBeDefined();
  });

  it('MainPage snapshot', () => {
    const mainpage = render(<MainPage />);
    expect(mainpage).toMatchSnapshot();
  });
});
