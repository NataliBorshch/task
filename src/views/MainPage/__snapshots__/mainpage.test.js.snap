// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component  MainPage render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main_page"
      >
        <div
          class="main_advice"
        >
          <h2>
            10 ways to make your day as productive as possible
          </h2>
          <ol>
            <li>
              <p>
                Every Sunday night, make plans for the week
              </p>
            </li>
            <li>
              <p>
                Set aside time in advance for specific tasks
              </p>
            </li>
            <li>
              <p>
                Make sure the list of tasks is achievable
              </p>
            </li>
            <li>
              <p>
                Set up a 30-minute default meeting
              </p>
            </li>
            <li>
              <p>
                Say no multitasking
              </p>
            </li>
            <li>
              <p>
                Try to make the best use of «border» time
              </p>
            </li>
            <li>
              <p>
                Watch the time
              </p>
            </li>
            <li>
              <p>
                Think about dinner
              </p>
            </li>
            <li>
              <p>
                Don’t take up any of the time you have to spend with your family
              </p>
            </li>
            <li>
              <p>
                Start every day right
              </p>
            </li>
          </ol>
        </div>
        <div
          class="main_img"
        >
          <img
            alt="time"
            src="https://ik.imagekit.io/s2fpg15d4rx/time_NifncTLm-ND.jpg"
          />
          <img
            alt="glasses"
            src="https://ik.imagekit.io/s2fpg15d4rx/glasses1_4_fSdw54Wgo.jpg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main_page"
    >
      <div
        class="main_advice"
      >
        <h2>
          10 ways to make your day as productive as possible
        </h2>
        <ol>
          <li>
            <p>
              Every Sunday night, make plans for the week
            </p>
          </li>
          <li>
            <p>
              Set aside time in advance for specific tasks
            </p>
          </li>
          <li>
            <p>
              Make sure the list of tasks is achievable
            </p>
          </li>
          <li>
            <p>
              Set up a 30-minute default meeting
            </p>
          </li>
          <li>
            <p>
              Say no multitasking
            </p>
          </li>
          <li>
            <p>
              Try to make the best use of «border» time
            </p>
          </li>
          <li>
            <p>
              Watch the time
            </p>
          </li>
          <li>
            <p>
              Think about dinner
            </p>
          </li>
          <li>
            <p>
              Don’t take up any of the time you have to spend with your family
            </p>
          </li>
          <li>
            <p>
              Start every day right
            </p>
          </li>
        </ol>
      </div>
      <div
        class="main_img"
      >
        <img
          alt="time"
          src="https://ik.imagekit.io/s2fpg15d4rx/time_NifncTLm-ND.jpg"
        />
        <img
          alt="glasses"
          src="https://ik.imagekit.io/s2fpg15d4rx/glasses1_4_fSdw54Wgo.jpg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MainPage component  MainPage snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main_page"
      >
        <div
          class="main_advice"
        >
          <h2>
            10 ways to make your day as productive as possible
          </h2>
          <ol>
            <li>
              <p>
                Every Sunday night, make plans for the week
              </p>
            </li>
            <li>
              <p>
                Set aside time in advance for specific tasks
              </p>
            </li>
            <li>
              <p>
                Make sure the list of tasks is achievable
              </p>
            </li>
            <li>
              <p>
                Set up a 30-minute default meeting
              </p>
            </li>
            <li>
              <p>
                Say no multitasking
              </p>
            </li>
            <li>
              <p>
                Try to make the best use of «border» time
              </p>
            </li>
            <li>
              <p>
                Watch the time
              </p>
            </li>
            <li>
              <p>
                Think about dinner
              </p>
            </li>
            <li>
              <p>
                Don’t take up any of the time you have to spend with your family
              </p>
            </li>
            <li>
              <p>
                Start every day right
              </p>
            </li>
          </ol>
        </div>
        <div
          class="main_img"
        >
          <img
            alt="time"
            src="https://ik.imagekit.io/s2fpg15d4rx/time_NifncTLm-ND.jpg"
          />
          <img
            alt="glasses"
            src="https://ik.imagekit.io/s2fpg15d4rx/glasses1_4_fSdw54Wgo.jpg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main_page"
    >
      <div
        class="main_advice"
      >
        <h2>
          10 ways to make your day as productive as possible
        </h2>
        <ol>
          <li>
            <p>
              Every Sunday night, make plans for the week
            </p>
          </li>
          <li>
            <p>
              Set aside time in advance for specific tasks
            </p>
          </li>
          <li>
            <p>
              Make sure the list of tasks is achievable
            </p>
          </li>
          <li>
            <p>
              Set up a 30-minute default meeting
            </p>
          </li>
          <li>
            <p>
              Say no multitasking
            </p>
          </li>
          <li>
            <p>
              Try to make the best use of «border» time
            </p>
          </li>
          <li>
            <p>
              Watch the time
            </p>
          </li>
          <li>
            <p>
              Think about dinner
            </p>
          </li>
          <li>
            <p>
              Don’t take up any of the time you have to spend with your family
            </p>
          </li>
          <li>
            <p>
              Start every day right
            </p>
          </li>
        </ol>
      </div>
      <div
        class="main_img"
      >
        <img
          alt="time"
          src="https://ik.imagekit.io/s2fpg15d4rx/time_NifncTLm-ND.jpg"
        />
        <img
          alt="glasses"
          src="https://ik.imagekit.io/s2fpg15d4rx/glasses1_4_fSdw54Wgo.jpg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
